src/test/java/apitests/DriverTest.java

```diff
@@ -0,0 +1,195 @@
+package apitests;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jdbc.demo.AddressDAO;
+import com.jdbc.demo.Application;
+import com.jdbc.demo.DriverDAO;
+import com.jdbc.demo.domain.psql.Address;
+import com.jdbc.demo.domain.psql.Driver;
+import utils.TestModelsFactory;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = Application.class)
+@WebIntegrationTest
+public class DriverTest {
+
+	private MockMvc mockMvc;
+
+	@Autowired
+	private DriverDAO driverManager;
+
+	@Autowired
+	private AddressDAO addressManager;
+
+	private ArrayList<Driver> driverList = new ArrayList<Driver>();
+	private ArrayList<Address> addressList = new ArrayList<Address>();
+
+	@Autowired
+	private WebApplicationContext webApplicationContext;
+
+	List<Driver> drivers;
+
+	Driver driver1;
+	Driver driver2;
+	Driver driver3;
+
+	private MediaType contentType = new MediaType(MediaType.APPLICATION_JSON.getType(),
+			MediaType.APPLICATION_JSON.getSubtype(), Charset.forName("utf8"));
+
+	@Before
+	public void setup() throws Exception {
+
+		mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+		this.addressList.add(addressManager.add(TestModelsFactory.createTestAddress1()));
+		driver1 = driverManager.add(TestModelsFactory.createTestDriver1(addressList.get(0)));
+		driverList.add(driver1);
+		driver2 = driverManager.add(TestModelsFactory.createTestDriver2(addressList.get(0)));
+		driverList.add(driver2);
+		drivers = driverManager.getAll();
```

Comment:
Since `driver3` is not added to database in `setup()` method ( in contradiction to `driver1` and `driver2`), perhaps it should have more distinct name.